src/zepto.js

```diff
@@ -622,9 +624,11 @@ var Zepto = (function() {
     },
     attr: function(name, value){
       var result
+
       return (typeof name == 'string' && !(1 in arguments)) ?
-        (!this.length || this[0].nodeType !== 1 ? undefined :
-          (!(result = this[0].getAttribute(name)) && name in this[0]) ? this[0][name] : result
+        (!this.length || this[0].nodeType !== 1 ? undefined :(
+          booleanAttributesRE.test(name) ? this[0][name] :
```

Comment:
Why did you choose to return a boolean value from specific properties? jQuery doesn't seem to do it:

```js
$('<input type=checkbox checked>').attr('checked')  //=> "checked"
```